internal/clients/team4/speaker.go

```diff
@@ -41,13 +43,63 @@ func (s *speaker) getActionsCost(actions []string) shared.Resources {
 }
 
 func (s *speaker) getHigherPriorityActionsCost(baseaction string) shared.Resources {
-	actionindex := len(SpeakerActionPriorities)
-	for i, action := range SpeakerActionPriorities {
+	actionindex := len(SpeakerActionOrder)
+	for i, action := range SpeakerActionOrder {
 		if action == baseaction {
 			actionindex = i
 		}
 	}
-	return s.getActionsCost(SpeakerActionPriorities[:actionindex])
+	priorityindex := len(SpeakerActionPriorities)
+	for i, action := range SpeakerActionPriorities {
+		if action == baseaction {
+			priorityindex = i
+		}
+	}
+	var SAPcopy []string = SpeakerActionPriorities[:0]
+	for _, action1 := range SpeakerActionPriorities[:priorityindex] {
+		alreadyExecuted := false
+		for _, action2 := range SpeakerActionOrder[:actionindex] {
+			if action1 == action2 {
+				alreadyExecuted = true
+			}
+		}
+		if !alreadyExecuted {
+			SAPcopy = append(SAPcopy, action1)
+		}
+	}
+	return s.getActionsCost(SAPcopy)
```

Comment:
Could you break down each step and explain what you are trying to achieve? :)

Reply:
46~51: find where the action is in the order
52~57: find where the action is in the priorities
58: initialise return slice
59: get slice of higher priority action
60~69: add to return slice if the higher priority action has not been executed yet

Reply:
I meant as comments in the code :P

Reply:
oh
good point